internal/mode/static/state/graph/policies.go

```diff
@@ -214,6 +236,48 @@ func processPolicies(
 	return processedPolicies
 }
 
+// checkForRouteOverlap checks if any route references the same hostname:port/path combination
+// as a route referenced in a policy.
+func checkForRouteOverlap(route *L7Route, hostPortPaths map[string]string) *conditions.Condition {
+	for _, parentRef := range route.ParentRefs {
+		if parentRef.Attachment != nil {
+			port := parentRef.Attachment.ListenerPort
+			for _, hostname := range parentRef.Attachment.AcceptedHostnames {
+				for _, rule := range route.Spec.Rules {
+					for _, match := range rule.Matches {
+						if match.Path != nil && match.Path.Value != nil {
```

Comment:
if Path is nil, don't we set it to `/`?

Reply:
No, nil path is an error condition on the route.